clients/js/src/Collection.ts

```diff
@@ -0,0 +1,573 @@
+     * @param {Metadata | Metadatas} [params.metadatas] - Optional metadatas to update.
+     * @param {Document | Documents} [params.documents] - Optional documents to update.
+     * @returns {Promise<boolean>} - The API Response. True if successful. Else, error.
+     *
+     * @example
+     * ```typescript
+     * const response = await collection.update({
+     *   ids: ["id1", "id2"],
+     *   embeddings: [[1, 2, 3], [4, 5, 6]],
+     *   metadatas: [{ "key": "value" }, { "key": "value" }],
+     *   documents: ["new document 1", "new document 2"],
+     * });
+     * ```
+     */
+    public async update({
+        ids,
+        embeddings,
+        metadatas,
+        documents,
+    }: {
+        ids: ID | IDs,
+        embeddings?: Embedding | Embeddings,
+        metadatas?: Metadata | Metadatas,
+        documents?: Document | Documents,
+    }): Promise<boolean> {
+        if (
+            embeddings === undefined &&
+            documents === undefined &&
+            metadatas === undefined
+        ) {
+            throw new Error(
+                "embeddings, documents, and metadatas cannot all be undefined"
+            );
+        } else if (embeddings === undefined && documents !== undefined) {
+            const documentsArray = toArray(documents);
+            if (this.embeddingFunction !== undefined) {
+                embeddings = await this.embeddingFunction.generate(documentsArray);
+            } else {
+                throw new Error(
+                    "embeddingFunction is undefined. Please configure an embedding function"
+                );
+            }
+        }
+
+        // backend expects None if metadatas is undefined
+        if (metadatas !== undefined) metadatas = toArray(metadatas);
+        if (documents !== undefined) documents = toArray(documents);
+
+        var resp = await this.api
+            .update(
+                this.id,
+                {
+                    ids: toArray(ids),
+                    embeddings: embeddings ? toArrayOfArrays(embeddings) : undefined,
+                    documents: documents,
+                    metadatas: metadatas
+                },
+            )
+            .then(handleSuccess)
+            .catch(handleError);
+
+        return resp;
+    }
+
+    /**
+     * Performs a query on the collection using the specified parameters.
+     *
+     * @param {Object} params - The parameters for the query.
+     * @param {Embedding | Embeddings} [params.query_embeddings] - Optional query embeddings to use for the search.
+     * @param {PositiveInteger} [params.n_results] - Optional number of results to return (default is 10).
+     * @param {Where} [params.where] - Optional query condition to filter results based on metadata values.
+     * @param {string | string[]} [params.query_text] - Optional query text(s) to search for in the collection (renamed to 'query_texts' in the future).
```

Comment:
let's go ahead and do this now https://github.com/chroma-core/chroma/pull/581

can you pull this change into your PR? ill close the other PR

Reply:
I totally missed these comments last night, on it now